### PR TITLE
(feat): Added hot reloading in our /classes page

### LINKED ISSUE #365: /classes should take advantage of fast-refresh.
**Describe the bug**
We currently are not taking advantage of refreshing our DOM when we create new classrooms and instead rely on reloading our page to see any changes in our UI. Making the data inside of our classes page rely on a state should help us solve that issue and remove the need to reload the page in order to see any new changes made to the UI.
**To Reproduce**
1. Go to `/classes`
2. Create, delete, or edit a classroom
3. Page must reload to see any changes to the DOM

**Expected behavior**
The DOM should refresh to show the new state of the `ClassInviteTable` component.

**Additional context**
Currently being worked on in #357

### PR DESCRIPTION
**Pull request by GuillermoFloresV**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Closes #365 

Our classes page is not currently taking advantage of hot reloading, this PR introduces changes to take advantage of this feature. 

Once we load our `/classes` page, we save them into a state. This state will help us keep track of our classes (adding, editing, deleting)

We are using our response to build the class into our DOM to ensure that the server and client are always matched together.

Here are GIF comparisons of the two experiences:

<details><summary>Before</summary>
<p>

**Creating a classroom:**

old-classes

**Deleting a classroom:**

old-deletion

</p>
</details> 

<details><summary>After</summary>
<p>

**Creating a classroom:**

new-classes-fast-refresh

**Deleting a classroom:**

deletion-fast-refresh

</p>

</details>

### CODE REVIEW

**Comment by lloydchang** [comment]
@Komal914 See above.

**Comment by Komal914** [comment]
@lloydchang yes, I think this may be what is causing the Alerts to disappear, since the page refreshes right after

**Comment by lloydchang** [comment]
Status: @Komal914 confirmed that this #357 is blocking #134

On July 17th, Komal Kaur messaged to Lloyd Chang, Guillermo Flores V in CodeDay Labs #help-desk channel at https://codedayorg.slack.com/archives/C05EZKUU456/p1689623550089979?thread_ts=1689278492.275899

> I pulled in the commits from fast-reload-classes and the alerts are staying up for the full 5 seconds :slightly_smiling_face:
>
> Attached: Screen Recording 2023-07-17 at 12.52.04 PM.mov

Guillermo Flores V replied:
> Great, that pr should hopefully be merged soon!

**Comment by GuillermoFloresV** [comment]
> Status: @Komal914 confirmed that this #357 is blocking #134
> 
> On July 17th, Komal Kaur messaged Lloyd Chang, Guillermo Flores V in CodeDay Labs #help-desk channel at https://codedayorg.slack.com/archives/C05EZKUU456/p1689623550089979?thread_ts=1689278492.275899
> 
> > I pulled in the commits from fast-reload-classes and the alerts are staying up for the full 5 seconds :slightly_smiling_face:
> >
> > Attached: Screen Recording 2023-07-17 at 12.52.04 PM.mov
> 
> Guillermo Flores V replied:
> > Great, that pr should hopefully be merged soon!

My PR is currently in review, sorry about the delay but I cannot merge it myself since I am the PR author.

Once the PR is merged we can move to reviewing Komal's PR

File: components/ClassInviteTable.js
Comment (utsab) [inline components/ClassInviteTable.js:57]: In the event that the delete operation fails, the browser will falsely indicate that the operation was successful.  The code does catch one failure case (403 error), but there might be other error codes not accounted for (such as a 500 error). The ideal solution I believe is to change the if statement to be the following: 

if (res.status !== 200) {
  // handle error 
}

I think it's ok to merge your PR anyway, but we should log this edge case for a future issue.

**Review by utsab — APPROVED** [review]

**Pull request merged**
